### PR TITLE
Change directory used by ssm commands

### PR DESCRIPTION
**Pull request by mrajashree**
*Issue #, if available:*

*Description of changes:*
E2e tests use ssm commands for running tests, which includes downloading
the required binaries and running the command for e2e tests.
SSM by default starts a session with /usr/bin path. This is why all binaries
and test folder created by an e2e test gets saved under that path.
This commit adds a command to change directory to a new path /home/e2e
before running any other commands so that the generated test output is saved
under /home/e2e and not under /usr/bin

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/ssm/command.go
Comment (pokearu) [inline internal/pkg/ssm/command.go:17]: Is there any specific reason we want to do this?
Because the binaries get [downloaded](https://github.com/aws/eks-anywhere/blob/main/internal/test/e2e/setup.go#L127) into /bin and I think /bin is referenced in a bunch of places.

Comment (mrajashree) [reply internal/pkg/ssm/command.go:17]: We just don't want the folder for the test containing all kubeconfig/cluster.yml files and generated logs to be created under `/usr/bin`, binaries can be under /bin...since many places refer /bin, I thought it'd be easier to create a bin folder under the new path, with this change all the test specific binaries will get downloaded under this /home/e2e/bin path. And the tests already [append the working directory to the path](https://github.com/aws/eks-anywhere/blob/main/test/framework/e2e.go#L203) so these binaries get used by the tests

Comment (mrajashree) [reply internal/pkg/ssm/command.go:17]: We don't want the folder for the test containing all kubeconfig/cluster.yml files and generated logs to be created under `/usr/bin`, nothing against dowloading binaries under /bin although it should be `/usr/local/bin` instead...since many places refer /bin, and the tests append the [`/working-directory/bin` to the path](https://github.com/aws/eks-anywhere/blob/main/test/framework/e2e.go#L203), I thought it'd be easier to create a bin folder under the new path, with this change all the test specific binaries will get downloaded under this /home/e2e/bin path and get used by the tests.

**Comment by abhay-krishna** [comment]
@mrajashree: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-cluster-controller-tooling-presubmit | 9967114250ac994e1828c8d1714c3ebc8cca3e66 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/228/eks-anywhere-cluster-controller-tooling-presubmit/1441089055935696896) | `/test eks-anywhere-cluster-controller-tooling-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=228). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:mrajashree).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Review by jaxesn — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/228#pullrequestreview-762604862" title="LGTM">jaxesn</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/228#" title="Author self-approved">mrajashree</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [jaxesn,mrajashree]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
